### PR TITLE
Don't send remote-dev mode password over the wire

### PR DESCRIPTION
**Pull request by stuartwdouglas**

### CODE REVIEW

File: docs/src/main/asciidoc/maven-tooling.adoc [hunk 1]
```diff
@@ -193,6 +193,11 @@ If you do not want to use the HTTP feature then you can simply run the `remote-d
 In this mode the command will continuously rebuild the local application, so you can use an external tool such as odo or
 rsync to sync to the remote application.
 
+NOTE: It is recommended you use SSL when using remote dev mode, however even if you are using an unencrypted connection
+your password is never sent directly over the wire. For the initial connection request the password is hashed with the
+initial state data, and subsequent requests hash it with a random session id generated by the server and any body contents
```
Comment (maxandersen) [inline docs/src/main/asciidoc/maven-tooling.adoc:198]: does that not mean if I stop the remote dev client side, do some edits and start again that I can't authenticate or do I misunderstand ?

Comment (stuartwdouglas) [reply docs/src/main/asciidoc/maven-tooling.adoc:198]: It creates a new session in that case.

File: extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/devmode/Hashes.java [hunk 2]
```diff
@@ -0,0 +1,26 @@
+package io.quarkus.vertx.http.runtime.devmode;
+
+import java.nio.charset.StandardCharsets;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+
+public class Hashes {
```
Comment (gsmet) [inline extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/devmode/Hashes.java:7]: We already have a `io.quarkus.deployment.util.HashUtil` around. Wondering if it makes sense to have everything in the same place?

(Moved to core runtime if the deployment artifact is not present here)

**Review by gsmet — APPROVED** [review]

**Comment by gsmet** [comment]
I just pushed a rebase to fix an import conflict.

**Pull request merged**